core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -444,17 +485,57 @@ ClientRequestContext newDerivedContext(RequestId id, @Nullable HttpRequest req,
      * @param responseTimeoutMillis the amount of time allowed in milliseconds from
      *                              the beginning of the response
      *
-     * @deprecated Use {@link #extendResponseTimeoutMillis(long)}, {@link #setResponseTimeoutAfterMillis(long)},
-     *             {@link #setResponseTimeoutAtMillis(long)} or {@link #clearResponseTimeout()}
      */
-    @Deprecated
-    void setResponseTimeoutMillis(long responseTimeoutMillis);
+    default void setResponseTimeoutMillis(long responseTimeoutMillis) {
+        setResponseTimeoutMillis(TimeoutMode.FROM_START, responseTimeoutMillis);
```

Comment:
Perhaps the sensible default would be `TimeoutMode.FROM_NOW`?

Reply:
Oops... thanks for pointing it out!🙏